server/apps/packages/takes_package_service.py

```diff
@@ -208,3 +212,58 @@ def get_first_take_in_takes_package(self, item):
                     return ref.get(ITEM_REF, None)
 
         return None
+
+    def __get_package_refs(self, package):
+        """
+        Get refs from the takes package
+        :param dict package: takes package
+        :return: return refs from the takes package. If not takes package or no refs found then None
+        """
+        refs = None
+        if package:
+            groups = package.get('groups', [])
+            refs = next((group.get('refs') for group in groups if group['id'] == MAIN_GROUP), None)
+
+        return refs
+
+    def can_publish_take(self, package, sequence):
+        """
+        Takes can be published only in ascending order. This function check that if there are any
+        unpublished takes before the current take.
+        :param dict package: takes packages
+        :param int sequence: take sequence of the published take
+        :return: True if takes are published in correct order else false.
+        """
+        refs = self.__get_package_refs(package)
+        if refs:
+            takes = [ref.get(ITEM_REF) for ref in refs if ref.get(SEQUENCE) < sequence]
+            # elastic filter for the archive resource filters out the published items
+            archive_service = get_resource_service(ARCHIVE)
+            query = {'query': {'filtered': {'filter': {'terms': {'_id': takes}}}}}
+            request = ParsedRequest()
+            request.args = {'source': json.dumps(query)}
+            items = archive_service.get(req=request, lookup=None)
+            return items.count() == 0
+
+        return True
+
+    def get_takes_in_take_package(self, takes_package):
```

Comment:
can we rename this method `get_published_takes()`? Looking at the method name gives an impression that it brings all the takes.